GitExtUtils/GitUI/ThreadHelper.cs

```diff
@@ -6,48 +6,26 @@
 using System.Runtime.InteropServices;
 using System.Threading;
 using System.Threading.Tasks;
-using System.Windows.Forms;
 using Microsoft.VisualStudio.Threading;
 
 namespace GitUI
 {
     public static class ThreadHelper
     {
         private const int RPC_E_WRONG_THREAD = unchecked((int)0x8001010E);
-        private static JoinableTaskContext _joinableTaskContext = null!;
-        private static JoinableTaskCollection _joinableTaskCollection = null!;
-        private static JoinableTaskFactory _joinableTaskFactory = null!;
+
+        private static TaskManager _taskManager;
 
         public static JoinableTaskContext JoinableTaskContext
         {
-            get
-            {
-                return _joinableTaskContext;
-            }
-
-            internal set
-            {
-                if (value == _joinableTaskContext)
-                {
-                    return;
-                }
-
-                if (value is null)
-                {
-                    _joinableTaskContext = null!;
-                    _joinableTaskCollection = null!;
-                    _joinableTaskFactory = null!;
-                }
-                else
-                {
-                    _joinableTaskContext = value;
-                    _joinableTaskCollection = value.CreateCollection();
-                    _joinableTaskFactory = value.CreateFactory(_joinableTaskCollection);
-                }
-            }
+            get => _taskManager?.JoinableTaskContext;
+            internal set => _taskManager = value is null ? null : new(value);
```

Comment:
```suggestion
            internal set => _taskManager = value ?? new(value);
```

Reply:
Since we're in a static context, we need to synchronise to avoid race conditions.

Reply:
I think `value ?? new(value)` would do the opposite of what is needed. We need to keep `null` or else to create the `TaskManager` if we have a context. A `TaskManager` always has a context.

> Since we're in a static context, we need to synchronise to avoid race conditions.

This property must be set and reset in a controlled way - and that's the case: set once on program start, reset synchronously in some tests.
So I think this is one of the rare cases where we do not need bother.

Reply:
Ah, yes. My mistake.